### DISCUSSION TITLE
Time series classification on panel data with cross validation

### DISCUSSION THREAD

**Discussion by bdeck8317** [alan-turing-institute/sktime#7507] [Q&A]
I have a panel dataset with multiple subjects with varying rows (time points). This is a binary classification task. I want to employ a nested cross-validation approach using a walk-forward validation (expanding window). I looked through the documentation on the ExpandingWindowSplitter and TSCGridSearchCV, yet it wasn't obvious to me how I should format my data. Should I create a multi-index data frame with my subject identifier and time stamp as the indices? Do these indices need to be present in my feature and label data frames? If I have subjects with varying time-sequences, how would I set up the ExpandingWindowSplitter for cross-validation? If I use ExpandingWindowSplitter, can I use that as my CV arg in TSCGridSearchCV?

Cheers!

**Comment by fkiraly** [comment]
@bdeck8317, can you explain what you mean by walk-forward validation? In time series classification, you have a single label per entire time series, so what exactly would you train the classifier on?

**Reply by bdeck8317** [reply]
Perhaps I am a confused...

I am attempting to classify people at risk for a disorder. My training data is longitudinal. In production the data will accumulate per subject. I need to make a prediction on these individuals at specific intervals, i.e., after every 4 hours. My thought was to use a walk-forward validation approach to select the best hyperparameters and make positive identification of risk for a disorder. This way, the training emulates the production environment.
